### PR TITLE
chore: Shopify Storefront API version 2022-10

### PR DESCRIPTION
**Pull request by NWRichmond**
## Why are these changes introduced?

Addesses [ENG-7394](https://nacelle.atlassian.net/browse/ENG-7394)

> Update to use 2022-10 version of the Shopify Storefront API

## What is this pull request doing?

- Updates `@nacelle/shopify-cart` to use the latest (`2022-10`) version of the Shopify Storefront API.
- Updates the lockfile for the Next.js Reference Store. This happened automatically when running `npm run dev` in that project (`next` detects lockfile issues and corrects them automatically).

## How to test

1. Check out the `ENG-7394-shopify-storefront-api-version-2022-10` branch.
2. `npm run bootstrap` in the monorepo root.
3. Navigate to `packages/shopify-cart`.
4. `npm run build` - the build should produce no errors.
5. `npm run test:ci` - all tests should pass with 100% coverage.

## Local testing results

In the Next.js Reference Store, the cart functioned as expected using the locally-bootstrapped `@nacelle/shopify-cart`:

https://user-images.githubusercontent.com/5732000/197248133-7a3cd3ab-bcc1-41e1-8a30-17569943846b.mov

### CODE REVIEW

File: packages/shopify-cart/src/types/shopify.type.ts [hunk 1]
```diff
@@ -911,32 +1033,32 @@ export type Checkout = Node & {
...
-   * The sum of all the prices of all the items in the checkout, taxes and discounts included.
-   * @deprecated Use `totalPriceV2` instead.
+   * The sum of all the prices of all the items in the checkout, including taxes and duties.
+   * @deprecated Use `totalPrice` instead.
    */
-  totalPrice: Scalars['Money'];
-  /** The sum of all the prices of all the items in the checkout, including duties, taxes, and discounts. */
   totalPriceV2: MoneyV2;
+  /** The sum of all the taxes applied to the line items and shipping lines in the checkout. */
+  totalTax: MoneyV2;
```
Comment (mdarrik) [inline packages/shopify-cart/src/types/shopify.type.ts:1057]: Do any of these `thingv2` -> `thing` deprecations/changes affect any of our queries?

Comment (NWRichmond) [reply packages/shopify-cart/src/types/shopify.type.ts:1057]: I searched for any uses and couldn't find any. Glad you noticed this too!

File: packages/shopify-cart/src/types/shopify.type.ts [hunk 2]
```diff
@@ -5460,12 +5660,12 @@ export type ProductVariant = HasMetafields & Node & {
   availableForSale: Scalars['Boolean'];
   /** The barcode (for example, ISBN, UPC, or GTIN) associated with the variant. */
   barcode?: Maybe<Scalars['String']>;
+  /** The compare at price of the variant. This can be used to mark a variant as on sale, when `compareAtPrice` is higher than `price`. */
+  compareAtPrice?: Maybe<MoneyV2>;
```
Comment (mdarrik) [inline packages/shopify-cart/src/types/shopify.type.ts:5664]: Looks like we're using a few of the PriceV2 and associated fields in the [merchandise fragment](https://github.com/getnacelle/nacelle-js/blob/95977b55a7faa099a662781c3327220b6d0e824e/packages/shopify-cart/src/graphql/fragments/merchandise.ts#L4)

Comment (NWRichmond) [reply packages/shopify-cart/src/types/shopify.type.ts:5664]: Good catch! Thanks. Fixed in https://github.com/getnacelle/nacelle-js/pull/280/commits/097927b515357ba577e47d478d1f3ad9e3030855.

**Review by mdarrik — APPROVED** [review]
LGTM - looks like you got all the v2 deprecated types, tests all pass, and I was able to add items to the cart and checkout in the Gatsby reference store.

**Pull request merged**